File: backend/Models/Category.cs
using System.Text.Json.Serialization;

namespace Backend.Models;

public class Category
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public int? ParentId { get; set; }

    // Navigation — excluded from JSON to avoid circular refs and empty arrays
    [JsonIgnore]
    public Category? Parent { get; set; }
    [JsonIgnore]
    public ICollection<Category> Children { get; set; } = [];
    [JsonIgnore]
    public ICollection<Photo> Photos { get; set; } = [];
}

File: frontend/src/types/category.ts
export interface Category {
  id: number
  name: string
  description?: string
  parentId?: number | null
}

export interface CategoryNode extends Category {
  children: CategoryNode[]
  photoCount?: number
}

export interface CategoryCreatePayload {
  name: string
  parentId?: number | null
  description?: string
}
